### ISSUE TITLE
Error while setting up ebeco platform for climate

### ISSUE THREAD

**Issue by misipakko** [joggs/home_assistant_ebeco#26]
I got this error after reboot.
Integration is dead. Worked before on same core version.

Runnig HA Core 2023.2.5 log:

2023-02-21 17:06:50.725 ERROR (MainThread) [homeassistant.components.climate] Error while setting up ebeco platform for climate
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 293, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/ebeco/climate.py", line 38, in async_setup_entry
    dev.append(EbecoClimateDevice(instance, device_data, sensor))
  File "/config/custom_components/ebeco/climate.py", line 47, in __init__
    super().__init__(instance, device_data["id"], main_sensor)
TypeError: 'NoneType' object is not subscriptable
2023-02-21 17:06:50.726 ERROR (MainThread) [homeassistant.components.sensor] Error while setting up ebeco platform for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 293, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/ebeco/sensor.py", line 32, in async_setup_entry
    dev.append(EbecoRelaySensor(instance, device_data, sensor))
  File "/config/custom_components/ebeco/sensor.py", line 44, in __init__
    super().__init__(instance, device_data["id"], sensor)
TypeError: 'NoneType' object is not subscriptable

**Comment by fijuval** [comment]
Same problem. I have been using this for a long time. Suddenly at 11:00 EET today it just died.

**Comment by joggs** [comment]
@TurboJonte or @PerfectlyNormal, do you have time to check what could be the problem? I will not be able to code for a while

**Comment by joggs** [comment]
Actually, I notice that the Ebeco integration cannot be accessed from Tibber either so Ebeco could have problems again.

**Comment by joggs** [comment]
I just spoke with a guy at Ebeco and apparently someone has hammered the api with requests so they had to disable the open api while trying to solve the issue, so this will probably resolve itself when the api is enabled again

**Comment by jonasgustavsson** [comment]
This is strange. What I noticed happening:
1. Yesterday at 10:15 EET, my automation for monitoring the temperature triggered an error due to my Ebeco thermostat becamoming unavailable.
2. A few hours later, I checked the Ebeco (native) Android app, and it seemed working just fine. 
3. A few hours later, the app started, but claimed my Ebeco device wasn't connected to my wifi.  
4. Today, when I open the Ebeco app, it doesn't even start properly, but just gives me an error "A problem occured while trying to communicate with the server".

Seems like they are doing different changes on their end, not only blocking the open API. It also indicates that their app uses the open API, which is good for us in the long run. Lets hope it's not a severe security incident, though.
